pipert/contrib/video_writer.py

```diff
@@ -0,0 +1,77 @@
+import cv2
+from pipert import BaseComponent, Routine
+from queue import Queue
+import argparse
+from urllib.parse import urlparse
+from imutils import resize
+from pipert.core.mini_logics import MessageFromRedis
+from pipert.core import QueueHandler, Message
+
+
+class VideoWriterLogic(Routine):
+
+    def __init__(self, output_file, in_queue, fps=30, im_size=(640, 480), *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.writer = None
+        self.output_file = output_file
+        self.q_handler = QueueHandler(in_queue)
+        self.w, self.h = im_size
+        self.fps = fps
+
+    def main_logic(self, *args, **kwargs):
+        msg: Message = self.q_handler.non_blocking_get()
+        if msg:
+            frame = msg.get_payload()
+            if frame.shape[0] != self.h or frame.shape[1] != self.w:
+                self.writer.release()
+                self.writer = cv2.VideoWriter(self.output_file, cv2.VideoWriter_fourcc('M', 'J', 'P', 'G'), self.fps,
+                                              (frame.shape[1], frame.shape[0]))
+                self.h, self.w = frame.shape[0], frame.shape[1]
+
+            frame = cv2.putText(frame, f"{msg.id.split('_')[-1]}", (0, self.h - 10), cv2.FONT_HERSHEY_SIMPLEX,
+                                1, (0, 0, 255), 2, cv2.LINE_AA)
+            print(frame.shape)
+            self.writer.write(frame)
+
+    def setup(self, *args, **kwargs):
+        self.writer = cv2.VideoWriter(self.output_file, cv2.VideoWriter_fourcc('M', 'J', 'P', 'G'), self.fps,
+                                      (self.w, self.h))
+
+    def cleanup(self, *args, **kwargs):
+        self.writer.release()
+
+
+class VideoWriter(BaseComponent):
+
+    def __init__(self, endpoint, in_key, redis_url, output_file, fps=30, im_size=(640, 480), name="VideoWriter"):
+        super().__init__(endpoint, name)
+        self.queue = Queue(maxsize=10)
+        t_stream = VideoWriterLogic(output_file, self.queue, fps, im_size, name="capture_frame",
+                                    component_name=self.name).as_thread()
+        t_stream.pace(30)
+        self.register_routine(t_stream)
+        t_upload = MessageFromRedis(in_key, redis_url, self.queue, most_recent=False, name="upload_redis",
+                                    component_name=self.name).as_thread()
+        self.register_routine(t_upload)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--in-key', help='Input key', type=str, default='camera:0')
+    parser.add_argument('-o', '--output', help='Output file name', type=str, default='output.avi')
+    parser.add_argument('-u', '--url', help='Redis URL', type=str, default='redis://127.0.0.1:6379')
```

Comment:
Use the redis_url env by the docker-compose:
import os
os.environ.get('REDIS_URL')

Reply:
OK

Reply:
Well, maybe [not OK...](https://github.com/docker/compose/issues/6691). We should look into that or replace docker-compose...

Anyway, this could be optional, if "REDIS_URL" is in the environment variables it could override the args.